src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,193 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int NUMBER_ONE = 1;
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        addToLastPosition(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == 0) {
+            addToFirstPosition(value);
+        } else if (index == size) {
+            addToLastPosition(value);
+        } else {
+            addInsideList(index, value);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            addToLastPosition(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexForGet(index);
```

Comment:
You already checked index in getNodeByIndex(index); method, so you can remove this line